src/writer.jl

```diff
@@ -1,398 +1,277 @@
-#################################################################
-# This file implements a .pomdpx file generator using the
-# POMDPs.jl interface.
-#################################################################
-
 abstract type AbstractPOMDPXFile end
 
-mutable struct POMDPXFile <: AbstractPOMDPXFile
-    file_name::AbstractString
-    description::AbstractString
+@with_kw struct POMDPXFile <: AbstractPOMDPXFile
+    filename::String
+    description::String = "This is a POMDPX file for a POMDP"
 
-    state_name::AbstractString
-    action_name::AbstractString
-    reward_name::AbstractString
-    obs_name::AbstractString
+    s_var_name::String = "state"
+    a_var_name::String = "action"
+    o_var_name::String = "observation"
+    r_var_name::String = "reward"
 
-    initial_belief::Vector{Float64}
+    s_name::Function = normalize
```

Comment:
One other thing that I just thought about s_name is a field with an abstract type, which can cause performance issues: https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type (and writing POMDPXFiles is sometimes the slowest part of solving a problem with SARSOP.jl, so performance will be important. I suggest using the pattern suggested in that link.